### ISSUE TITLE
Can AvroDeserializer handle nullable fields?

### ISSUE THREAD

**Issue by ipinak** [confluentinc/confluent-kafka-dotnet#608]
Description
===========
Some of the messages I receive from Kafka have null values. Can this be handler by `AvroDeserializer`?

Before using AvroDeserializer I used microsoft's avro deserializer, but it is very slow and I want to switch to something better, that's why I am trying this implementation.

We receive the following message and I assume that some field is null, thus the exception.

```
public SpecificDeserializerImpl(ISchemaRegistryClient schemaRegistryClient)
{
    ...
    else
    {
         throw new ArgumentException(
             $"{nameof(AvroDeserializer<T>)} " +
            "only accepts type parameters of int, bool, double, string, float, " +
            "long, byte[], instances of ISpecificRecord and subclasses of SpecificFixed.");
    }
}
```

Some extra information

```
"Error":{
  "Code":"Local_ValueDeserialization",
  "Reason":"System.ArgumentException: AvroDeserializer only accepts type parameters of int, bool, double, string, float, long, byte[], instances of ISpecificRecord and subclasses of SpecificFixed.\n   at Confluent.Kafka.Serialization.SpecificDeserializerImpl`1..ctor(ISchemaRegistryClient schemaRegistryClient)\n   at Confluent.Kafka.Serialization.AvroDeserializer`1.Deserialize(String topic, Byte[] data)\n   at Confluent.Kafka.Serialization.MessageExtensions.Deserialize[TKey,TValue](Message message, IDeserializer`1 keyDeserializer, IDeserializer`1 valueDeserializer)",
  "HasError":true,
  "IsLocalError":true,
  "IsBrokerError":false
```

How to reproduce
================

Checklist
=========
Please provide the following information:

 - [x] Confluent.Kafka nuget version: 0.11.5
 - [ ] Apache Kafka version: 
 - [ ] Client configuration:
 - [X] Operating system: CentOS 7
 - [ ] Provide logs (with "debug" : "..." as necessary in configuration)
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by mhowlett** [comment]
handling of `null` values is implemented in the `1.0-experimental-11` package. It hasn't been back ported to 0.11.x.

**Comment by mhowlett** [comment]
beta is fairly imminent. a full release will probably be a few months after (the reason for the delay is we want to give the API changes a chance to bake-in before we lock them down).
